### ISSUE TITLE
--loop 0 results in full-wire speed rather than pcap speed.

### ISSUE THREAD

**Issue by fklassen** [appneta/tcpreplay#191]
... as per mail list

```
I am currently seeing this issue running both
tcpreplay version: 4.1.1-beta1 (build git:v4.1.0-18-gae55544)
and
tcpreplay version: 4.1.0 (build git:v4.1.0)

CentOS Linux release 7.1.1503 (Core)
Linux dev 3.10.0-229.1.2.el7.x86_64 #1 SMP Fri Mar 27 03:04:26 UTC
2015 x86_64 x86_64 x86_64 GNU/Linux

My current work around is to do the looping in the shell

while true ; do /usr/local/bin/tcpreplay -T nano -x 1 -i em1 capture.pcap ; done

but I'd prefer get back to letting tcpreplay do the looping.  Any
suggestions for alternate switches to try?  Or anything I can do to
help debug?
```

**Comment by gmorganVA** [comment]
I am seeing a similar problem that I haven't fully characterized yet.  I am running 4.1.0 + a patch to resolve the timestamps going back in time looping issue.  If I run -l 0 without limiting playback speed then tcpreplay crashes after playback of approximately 45 minutes at 1 Gb/s.  This is also on CentOS 7.1.1503 (Core) on x86_64.  I'd be happy to help debug also.

**Comment by gmorganVA** [comment]
The problem I found may not be related to the issue listed above.  There are several warnings in replay_two_files() in replay.c that are have a return code which does not indicate that they are warnings (needs to be rcode=1 in order for the code right after tcpreplay_replay(ctx) returns in the main() function in tcpreplay.c to properly process the warning (as opposed to thinking it is an error...all negative return codes are treated as errors).  This is in the source for 4.1.0. I haven't pulled a newer version yet.

**Comment by fklassen** [comment]
Thanks for the input. Best guess is that I will be working on this in September.

**Comment by gmorganVA** [comment]
Is there a reference for best practices in providing fixes for inclusion in the tree?  The test I was running previously would fail reliably in 45 minutes or so and with my current fix, the same test has been running for over three days straight.

**Comment by fklassen** [comment]
http://tcpreplay.appneta.com/wiki/contributions.html

**Comment by fklassen** [comment]
@gmorganVA I am about to start work on this soon. I have not reproduced it yet. I see that you have not forked the REPO yet, so I guess you may have questions about how to contribute. I would love to see your changes. If you still want to contribute, you can chat with me here https://www.hipchat.com/gn8S5TnbV

**Comment by fklassen** [comment]
Finally recreated this by accident. Discovered error happens if you specify --loop=X (anything other than 1) and no other options such as -M or -t. It is a combination I rarely use. I guess I didn't try those combos during regression testing. Should have a fix soon.

Too bad I didn't see the command-line options that produced the error. It would have been fixed much sooner.

**Comment by fklassen** [comment]
Fixed in PR #229

**Comment by gmorganVA** [comment]
Fred,

Thanks for the updates. I would have responded sooner but I no longer
have access to the source code where I was working on this bug since I
have recently changed places of employment. I will certainly contact my
old team and let them know of the developments because this is still
relevant to their work.

--
  George Morgan
 
 
On Fri, Dec 18, 2015, at 08:49 PM, Fred Klassen wrote:
> Fixed in PR #229[1]


> —
> Reply to this email directly or view it on GitHub[2]. 


>  
 


Links:

  1. https://github.com/appneta/tcpreplay/pull/229
  2. https://github.com/appneta/tcpreplay/issues/191#issuecomment-165934756
